enterprise/server/backends/distributed/distributed.go

```diff
@@ -142,12 +177,50 @@ func (c *Cache) GetMulti(ctx context.Context, digests []*repb.Digest) (map[*repb
 	return foundMap, nil
 }
 
-func (c *Cache) Set(ctx context.Context, d *repb.Digest, data []byte) error {
-	peer := c.peer(d)
-	if peer == c.myAddr {
-		return c.local.Set(ctx, d, data)
+type multiCloser struct {
+	closers []io.WriteCloser
+}
+
+func (mc *multiCloser) Write(data []byte) (int, error) {
+	for _, w := range mc.closers {
+		n, err := w.Write(data)
+		if err != nil {
+			return 0, err
+		}
+		if n != len(data) {
+			return n, io.ErrShortWrite
+		}
 	}
-	wc, err := c.cacheProxy.RemoteWriter(ctx, peer, c.prefix, d)
+	return len(data), nil
+}
+func (mc *multiCloser) Close() error {
+	for _, w := range mc.closers {
+		if err := w.Close(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *Cache) multiWriter(ctx context.Context, d *repb.Digest) (io.WriteCloser, error) {
+	peers := c.peers(d)
+	var wc io.WriteCloser
+	for _, peer := range peers {
+		rwc, err := c.cacheProxy.RemoteWriter(ctx, peer, c.prefix, d)
+		if err != nil {
+			return nil, err
+		}
+		if wc == nil {
+			wc = rwc
+		} else {
+			wc = &multiCloser{[]io.WriteCloser{rwc, wc}}
```

Comment:
looks like this builds a recursive tree of multiClosers, probably want a flat `closers` array instead?

Reply:
Done.